src/login.service.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  BadRequestException,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from './entities/user.entity';
+import { JWT, JWTResult } from './typings/jwt.typing';
+
+@Injectable()
+export class LoginService {
+  constructor(
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+    private jwtService: JwtService,
+  ) {}
+
+  async login(username: string, password: string): Promise<JWTResult> {
```

Comment:
Since you need to add `openapi` here, you need to move both of this params into a DTO `LoginDto`.

Reply:
Resolved on 44e1caa5eb347c93258954577b5d4c444fc0c829.